### ISSUE TITLE
PB626: News downloader - "Illegal instruction" causes KOReader crash

### ISSUE THREAD

**Issue by sonix-github** [koreader/koreader#5032]
* KOReader version: v2019.05-8-g34e6f41_2019-05-18 
* Device: PB626 (PBLux3) fw: W626.5.14.1225

#### Issue
After starting "Download news" it shows for a split of the second "processing..." and then KOReader crashes.

#### Steps to reproduce
Start KOReader, in upper menu select "settings", select News (RSS/Atom) downloader and tap "Download news". It starts to download feeds but immediately after showing "processing..." it crashes KOReader.

##### `crash.log` : [crash_news_debug_20190518_1319.log](https://github.com/koreader/koreader/files/3194008/crash_news_debug_20190518_1319.log)

**Comment by NiLuJe** [comment]
Is that feed available over plain HTTP? Does it crash then?

**Comment by NiLuJe** [comment]
(My train of thought being NEON SIMD is enabled in our OpenSSL builds on PB, as we're expecting a Cortex A8, which *should* hold true for anything since the i.MX50x, but PB being what it is, who knows...).

**Comment by sonix-github** [comment]
> Is that feed available over plain HTTP? Does it crash then?

@NiLuJe : I have changed https to http and it is still crashing. Before KOReader update I used some version from april 2019 and there was downloading ok.

**Comment by sonix-github** [comment]
Last functional KOReader version that I have on hand is from 11.3.2019 (git-rev: v2019.03.1"

**Comment by Thra11** [comment]
> Last functional KOReader version that I have on hand is from 11.3.2019 (git-rev: v2019.03.1"

Just to clarify, does that mean that 2019.04 also exhibits the crash, or that you didn't have it on hand to test?

**Comment by sonix-github** [comment]
Today I have tested nightly builds below and none of them except "v2019.03.1" is without crash. I do not have local copy of archives from 2019-03-11 to 2019-05-02 for comparison.

I have found that it has something with "epubdownloadbackend.lua" and "sink" table.

If I add, just before line _logger.dbg("request:", request)_ (which is the last debug message from crash log) line logger.dbg("table.getn(sink):", table.getn(sink)) I get value 0. I do not know if this is the right way to find the error but I do not have an idea how to debug it by myself.

> 05/20/19-20:30:46 DEBUG EpubDownloadBackend:getResponseAsString( https://www.kosmonautix.cz/feed/?cat=-1884 )
> 05/20/19-20:30:46 DEBUG getUrlContent( https://www.kosmonautix.cz/feed/?cat=-1884 ,
> 05/20/19-20:30:46 DEBUG timeout: 10
> **05/20/19-20:30:46 DEBUG table.getn(sink): 0**
> 05/20/19-20:30:46 DEBUG sink[1]:
> 05/20/19-20:30:46 DEBUG request: {
>     ["url"] = "https://www.kosmonautix.cz/feed/?cat=-1884",
>     ["method"] = "GET",
>     ["sink"] = function: 0x4550fa30
> }
> Illegal instruction

Versions that I have tried:
_koreader-pocketbook-arm-obreey-linux-gnueabi-v2019.04-78-g3009f5a_2019-05-03.zip
koreader-pocketbook-arm-obreey-linux-gnueabi-v2019.04-80-g4ba7f98_2019-05-05.zip
koreader-pocketbook-arm-obreey-linux-gnueabi-v2019.04-81-g08fa5e2_2019-05-06.zip
koreader-pocketbook-arm-obreey-linux-gnueabi-v2019.04-76-gaec7594_2019-05-02.zip
koreader-pocketbook-arm-obreey-linux-gnueabi-v2019.03.1.zip_
